validation/tmSchemaGenerator.js

```diff
@@ -110,9 +122,14 @@ function removeRequired(argObject) {
     // remove required if it exists and is of array type.
     // check for array is needed since we also specify what a required is and that it is an object
     if (("required" in argObject) && (Array.isArray(argObject.required))){
-        // need to decide whether to delete or replace it with ""
-        // delete is "cleaner" but "" is more explicit
-        delete argObject.required;
+        if (!(argObject["required"] == "sizes" || argObject["required"] == "rel")) {
```

Comment:
Could you add some comments on what is happening here?

Reply:
Sure! Will do :)

Reply:
I added a comment in c7b5fe5cbe9c24ad76c96c89a8fe15c305e53fb6